### DISCUSSION TITLE
Renaming file doesn't update header

### DISCUSSION THREAD

**Discussion by psarras** [SilentVoid13/Templater#632] [Help] [answered]
I've been trying to rename a new file I've created using `Create new note from template` and it works as expected, however the title on the header remains as `Untitled`. 

This is what I have:

```js
<%*
title = await tp.system.prompt("Title");

//const title = tp.file.title
const lower = title.toLowerCase()

if (lower.includes('weekly'))
{
	await tp.file.rename(title + " " + tp.date.now("YYYY-[W]W") );
	return tp.file.include('[[title]]');
} 
else
{
	await tp.file.rename(title + " " + tp.date.now("YYYY-MM-DD") );
	return tp.file.include('[[title]]');
}
%>
```
title.md
```
# <% tp.file.title %>
```

I am getting

<img width="249" alt="image" src="https://user-images.githubusercontent.com/3820417/167317934-11ac9efb-de4a-455b-a076-365c1aaba4db.png">

**Comment by AB1908** [comment]
Ah hmm seems like an order of operations thing. I'll test this out if I can.

**Reply by psarras** [reply]
I also noticed this: #522

**Reply by AB1908** [reply]
I forgot to check this out. I'm curious why you aren't using the variable from the prompt directly instead of `tp.file.title`.

**Reply by welpdx** [reply]
#522 was a good reference. 
Seems `tp.config.target_file.basename` and `this.app.workspace.getActiveFile().basename` both work better than `tp.file.title` right now. I think he answered his own question noticing that issue.

**Comment by welpdx** [comment]
I agree with @AB1908. 
I recommend using obsidian api in `file.md` for now. 
So instead of `<% tp.file.title %>`
try `<% this.app.workspace.getActiveFile().basename %>`

**Reply by psarras** [reply]
will try this today!

Ok this is working, not pretty but doesn't matter for a template :P

**Reply by welpdx** [reply]
:P

**Reply by pjar** [reply]
This workaround does not work for me either - it just puts the name of the template file but it ignores any file name changes.

**Reply by welpdx** [reply]
> This workaround does not work for me either - it just puts the name of the template file but it ignores any file name changes.

Hey @pjar, I can better help you if you make a help request with your template script and stuff. Thanks

**Reply by psarras** [reply]
Curiously this is not working if the active pane doesn't include a note! So if you close all your opened panes this will give an error `Cannot read properties of null (reading 'basename')` Any thoughts @welpdx ?

**Reply by AB1908** [reply]
Ya, you should be doing a null check first after storing it in a variable. 

```js
let curFile = \\command here;
if (curFile === null) return;
```

**Reply by welpdx** [reply]
You can use try catch:
```js
<%* try {var title = this.app.workspace.getActiveFile().basename; tR += title} catch(e) {console.log(e)} %>
```

**Reply by psarras** [reply]
Sorry I had to remove the answer, as this has since stopped working. 

Just to clarify

first of all thank you for the try / catch, however this is only fixing the issue of null if you don't have any document open. If you do have a document open, it provides you of header of this pane. Whereas, this was a fix for getting the header of the newly created document instead! ideally this should work: ```<% tp.file.title %>```

**Reply by welpdx** [reply]
try this in `title.md`. Forget about the try catch, tp.config should have no problem with that
```js
<% tp.config.target_file.basename %>
```

**Reply by psarras** [reply]
@welpdx Yep that worked! Do you want to add it as a separete answer, so I can accept it?

**Reply by welpdx** [reply]
Sure! Happy that it worked!

**Comment by welpdx** [comment, marked as answer]
<% tp.config.target_file.basename %>
